File: gastromanager-backend/src/main/java/com/kaiho/gastromanager/application/menu/mapper/MenuMapper.java
package com.kaiho.gastromanager.application.menu.mapper;

import com.kaiho.gastromanager.application.menu.dto.request.MenuRequestDto;
import com.kaiho.gastromanager.application.menu.dto.request.MenuUpdateRequestDto;
import com.kaiho.gastromanager.application.menu.dto.response.MenuDetailResponseDto;
import com.kaiho.gastromanager.application.menu.dto.response.MenuSummaryResponseDto;
import com.kaiho.gastromanager.domain.menu.model.Menu;
import com.kaiho.gastromanager.domain.restaurant.model.Restaurant;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import static com.kaiho.gastromanager.infrastructure.config.context.RestaurantContext.getCurrentRestaurant;

@Component
@RequiredArgsConstructor
public class MenuMapper {

    public Menu toDomain(MenuRequestDto dto) {
        return Menu.builder()
                   .name(dto.name())
                   .description(dto.description())
                   .isEnabled(true)
                   .restaurant(Restaurant.builder().uuid(getCurrentRestaurant()).build())
                   .build();
    }

    public MenuDetailResponseDto toResponseDetail(Menu menu) {
        return MenuDetailResponseDto.builder()
                                    .uuid(menu.getUuid())
                                    .name(menu.getName())
                                    .description(menu.getDescription())
                                    .isEnabled(menu.isEnabled())
                                    .createdBy(menu.getCreatedBy())
                                    .createdDate(menu.getCreatedDate())
                                    .updatedBy(menu.getUpdatedBy())
                                    .updatedDate(menu.getUpdatedDate())
                                    .build();
    }

    public MenuSummaryResponseDto toResponseSummary(Menu menu) {
        if (menu == null) {
            return null;
        }
        return MenuSummaryResponseDto.builder()
                                     .uuid(menu.getUuid())
                                     .name(menu.getName())
                                     .description(menu.getDescription())
                                     .createdBy(menu.getCreatedBy())
                                     .createdDate(menu.getCreatedDate())
                                     .updatedBy(menu.getUpdatedBy())
                                     .updatedDate(menu.getUpdatedDate())
                                     .build();
    }

    public Menu toDomain(MenuUpdateRequestDto menuRequestDto) {
        if (menuRequestDto == null) {
            return null;
        }
        return Menu.builder()
                   .name(menuRequestDto.name())
                   .description(menuRequestDto.description())
                   .restaurant(Restaurant.builder().uuid(getCurrentRestaurant()).build())
                   .build();
    }
}

File: gastromanager-frontend/src/app/features/menus/application/mappers/menu.mapper.ts
import {MenuResponseDto} from "../../domain/models/menu-response-dto.interface";
import {Menu} from "../../domain/models/menu.interface";
import {MenuRequestDto} from "../../domain/models/menu-request-dto.interface";
import {MenuDetailResponseDto} from "../../domain/models/menu-detail-response-dto.interface";

export function mapToMenuSummary(dto: MenuResponseDto): Menu {
  return {
    uuid: dto.uuid,
    name: dto.name,
    description: dto.description,
    isEnabled: dto.isEnabled,
  };
}

export function mapToMenuRequestDto(menu: Menu): MenuRequestDto {
  return {
    name: menu.name,
    description: menu.description,
  }
}

export function mapToMenuDetail(menuResponseDto: MenuDetailResponseDto): Menu {
  return {
    uuid: menuResponseDto.uuid,
    name: menuResponseDto.name,
    description: menuResponseDto.description,
    isEnabled: menuResponseDto.isEnabled,
    createdBy: menuResponseDto.createdBy,
    createdDate: new Date(menuResponseDto.createdDate),
    updatedBy: menuResponseDto.updatedBy,
    updatedDate: new Date(menuResponseDto.updatedDate),
  }
}

export function mapToEditMenuRequestDto(menu: Menu): MenuRequestDto {
  return {
    name: menu.name,
    description: menu.description,
  }
}


